app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,63 @@
+class PaymentsController < ApplicationController
+  before_action :set_payment, only: [:edit, :update, :destroy]
+  before_action :validates_role, only: [:new, :edit, :update, :create]
+  before_action :validates_admin, only: :destroy
+
+  def new
+    find_project
+    @payment = Payment.new
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
+    @payment.project_id = params[:project_id]
+
+    if @payment.save
+      redirect_to projects_path, notice: 'Payment was successfully created.'
+    else
+      find_project
+      render :new
+    end
+  end
+
+  def edit
+    find_project
+  end
+
+  def update
+    find_project
```

Comment:
so many find_projects in first line are shouting for them to be added as before action.

Reply:
though do we really need find project here?